src/App.vue

```diff
@@ -69,15 +75,68 @@ export default defineComponent({
 		});
 		let i: NodeJS.Timeout; // eslint-disable-line
 
-		// Fetch authed user
-		const { onResult } = useQuery<ClientRequiredData>(GetClientRequiredData);
-		onResult((res) => {
-			if (!res.data) {
-				return;
-			}
-			store.commit("SET_USER", res.data.clientUser);
-			store.commit("SET_GLOBAL_EMOTE_SET", res.data.globalEmoteSet);
-		});
+		// Set up client user
+		const stoppers = [] as (() => void)[]; // stop functions for out of context subscriptions
+		(async () => {
+			provideApolloClient(apolloClient);
+			const authToken = computed(() => store.getters.authToken);
+
+			await new Promise((resolve) => {
```

Comment:
what does this do:?
I think, this is not needed.

Reply:
like why run this in async?

Reply:
It's to wait for the user to sign in. Otherwise the query runs and returns `null` in currentUser, then signing in doesn't log the user in until next refresh